io_scene_nif/modules/property/material/material_import.py

```diff
@@ -162,62 +143,33 @@ def import_material(self, n_mat_prop, n_texture_prop,
-                b_mat.diffuse_intensity = 1.0
-
-            if (b_mat.diffuse_color.r + b_mat.diffuse_color.g + b_mat.diffuse_color.g) == 0:
-                b_mat.diffuse_color.r = bs_shader_property.hair_tint_color.r
-                b_mat.diffuse_color.g = bs_shader_property.hair_tint_color.g
-                b_mat.diffuse_color.b = bs_shader_property.hair_tint_color.b
-                b_mat.diffuse_intensity = 1.0
-
-            # Emissive
-            b_mat.niftools.emissive_color.r = bs_shader_property.emissive_color.r
-            b_mat.niftools.emissive_color.g = bs_shader_property.emissive_color.g
-            b_mat.niftools.emissive_color.b = bs_shader_property.emissive_color.b
-            b_mat.emit = bs_shader_property.emissive_multiple
-
-            # Alpha
-            if n_alpha_prop:
-                b_mat = self.set_alpha(b_mat, bs_shader_property, n_alpha_prop)
-
-            # gloss
-            b_mat.specular_hardness = bs_shader_property.glossiness
-
-            # Specular color
-            b_mat.specular_color.r = bs_shader_property.specular_color.r
-            b_mat.specular_color.g = bs_shader_property.specular_color.g
-            b_mat.specular_color.b = bs_shader_property.specular_color.b
-            b_mat.specular_intensity = bs_shader_property.specular_strength
-
-            # lighting effect
-            b_mat.niftools.lightingeffect1 = bs_shader_property.lighting_effect_1
-            b_mat.niftools.lightingeffect2 = bs_shader_property.lighting_effect_2
-
-        if n_mat_prop is None and bs_effect_shader_property:
-            # Alpha
-            if n_alpha_prop:
-                b_mat = self.set_alpha(b_mat, bs_shader_property, n_alpha_prop)
-
-            if bs_effect_shader_property.emissive_color:
-                b_mat.niftools.emissive_color.r = bs_effect_shader_property.emissive_color.r
-                b_mat.niftools.emissive_color.g = bs_effect_shader_property.emissive_color.g
-                b_mat.niftools.emissive_color.b = bs_effect_shader_property.emissive_color.b
-                b_mat.niftools.emissive_alpha = bs_effect_shader_property.emissive_color.a
-                b_mat.emit = bs_effect_shader_property.emissive_multiple
-            b_mat.niftools_alpha.textureflag = bs_effect_shader_property.controller.flags
-
         # check wireframe property
         if n_wire_prop:
             # enable wireframe rendering
             b_mat.type = 'WIRE'
 
-        self.nif_import.dict_materials[material_hash] = b_mat
+        self.dict_materials[material_hash] = b_mat
         return b_mat
+
+    def set_material_vertex_mapping(self, b_mesh, f_map, material, n_uvco):
+        if material:
+            # fix up vertex colors depending on whether we had textures in the material
+            mbasetex = self.texturehelper.has_base_texture(material)
+            mglowtex = self.texturehelper.has_glow_texture(material)
+            if b_mesh.vertex_colors:
```

Comment:
pretty sure that all of this is removed from 2.8, could mark as depreciated

Reply:
Cant wait